editor/layout/index.js

```diff
@@ -21,46 +22,55 @@ import VisualEditor from '../modes/visual-editor';
 import UnsavedChangesWarning from '../unsaved-changes-warning';
 import DocumentTitle from '../document-title';
 import AutosaveMonitor from '../autosave-monitor';
-import { removeNotice } from '../actions';
+import { removeNotice, toggleSidebar } from '../actions';
 import MetaBoxes from '../meta-boxes';
 import {
 	getEditorMode,
 	isEditorSidebarOpened,
 	getNotices,
 } from '../selectors';
 
-function Layout( { mode, isSidebarOpened, notices, ...props } ) {
-	const className = classnames( 'editor-layout', {
-		'is-sidebar-opened': isSidebarOpened,
-	} );
+class Layout extends Component {
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.isSidebarOpened && nextProps.isMobile && ! this.props.isMobile ) {
+			this.props.toggleSidebar();
```

Comment:
I wonder if we could achieve it with a Higher-order reducer instead.

```js
const myReducer = combineReducer( { ... } );

const myEnhancedReducer = ( state, action ) => {
  const newState = myReducer( state, action );
  
  if ( // check that isMobile and the previous state is not mobile using state and newSTate ) {  
     newState.preferences = {
        ...newState.preferences,
        isSidebarOpened: false, 
    };
 }

  return newState;
}
```

Do you think it's better this way? cc @aduth

Reply:
Hi @youknowriad, for now, a Higher-order reduce could be an option. But when implementing the more advanced logic proposed by @jasmussen, I'm not certain if a high order reducer makes things simpler. Because if we want to automatically open the sidebar again when increasing the size if it was opened before we decreased size we need some state information which I was thinking on adding to the state of layout component and just dispatch close /open sidebar when needed. With a hight order component, I think we need to add more information to the redux state which is not bad I think but I'm not certain if this information necessary for this logic should be there.

Reply:
I think if we want the proposed logic, we may need two reducer value. `isSidebarOpened` for desktop and `isSidebarOpenedOnMobile` which may simplify a lot of things :)

Reply:
Yes, that is a very nice way of handling the situation!

Reply:
Hi @youknowriad, to get a high order reducer here as it needs access to responsive and to preferences, it would need to be a top-level high order reducer, so it can access both responsive and preferences. It would be our first top-level high order reducer, is that right or I'm seeing something wrong?

Reply:
True, but the more I think about it, the more two reducer values seems good. which also solves the persistence issue, since we'd have to persist only one of these two preferences.

Reply:
Thank you for confirming my logic @youknowriad. Two reducer values sound like a good option for this case. But I think if we add an isSidebarOpenedMobile preference we still have the persistence issue. Because even the expected behavior of that ticket is for the sidebar to be closed on reload on mobile even if it was opened before. Our current store persists implementation stores all the preferences and we want it to be generic and not contain editor logic, so the best way to handle this problem I think is using the same logic we have now in PR https://github.com/WordPress/gutenberg/pull/3331 but switch to ignore isSidebarOpenedMobile instead of isSidebarOpened.

Also with this change, we would at least need to change the isSidebarOppened selector to check if we should use the normal sidebarOppened flag or the mobile one. In the toggling places, we would also need some logic to decide if we are toggling mobile sidebar or the normal one. I think this are the changes needed but I may be pursuing a more complex path.